examples/onnxruntime/inceptionV3/inceptionv3.py

```diff
@@ -0,0 +1,236 @@
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#####################################################################################
+# Inference with ONNX Runtime
+import onnxruntime
+import time
+from torchvision import models, transforms as T
+import torch
+from PIL import Image
+import numpy as np
+import argparse
+import os
+import subprocess
+
+#Use most upto date weights
+inception_v3 = models.inception_v3(weights=models.Inception_V3_Weights.DEFAULT, progress=True)
+
+# Download ImageNet labels
+#!curl -o imagenet_classes.txt https://raw.githubusercontent.com/pytorch/hub/master/imagenet_classes.txt
+
+def parse_input_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--fp16",
+        action="store_true",
+        required=False,
+        default=False,
+        help='Perform fp16 quantization on the model before running inference',
+    )
+
+    parser.add_argument(
+        "--image_dir",
+        required=False,
+        default="../resnet50/images",
+        help='Target DIR for images to infer. Default is ./images'
+    )
+
+    parser.add_argument(
+        "--batch",
+        required=False,
+        default=1,
+        help='Batch size of images per inference',
+        type=int
+    )
+
+    parser.add_argument(
+        "--top",
+        required=False,
+        default=1,
+        help='Show top K of inference results',
+        type=int
+    )
+
+    parser.add_argument(
+        "--QPS",
+        action="store_true",
+        required=False,
+        default=False,
+        help='Show inference result in Queries-Per-Second QPS instead of inference duration (milliseconds)',
+    )
+
+    parser.add_argument(
+        "--verbose",
+        action="store_true",
+        required=False,
+        default=False,
+        help='Show verbose output',
+    )
+    
+    return parser.parse_args()
+
+def get_image_list_in_dir(dir):
+    proc = subprocess.run("ls",
+                          shell=True,
+                          stdout=subprocess.PIPE,
+                          cwd=dir)
+    fileList = proc.stdout.decode().split('\n')
+    fileList
+    return fileList
+
+def softmax(x):
+    """Compute softmax values for each sets of scores in x."""
+    e_x = np.exp(x - np.max(x))
+    return e_x / e_x.sum()
+
+def run_sample(session, categories, latency, inputs, topk, batch_size):
```

Comment:
This method will include the host-device copy overheads. There is some [documentation](https://onnxruntime.ai/docs/api/python/api_summary.html#data-on-device:~:text=The%20input%20data%20and%20output%20data%20are%20both%20on%20a%20device%2C%20users%20directly%20use%20the%20input%20and%20also%20place%20output%20on%20the%20device) for leaving intputs and outputs on host.

Reply:
Ah I thought timing would not include that and the copy happens after the bind.

Reply:
ahh i missed that explicit output copy. Rereading the docs, Im not sure this will include the host->device copy for the input though. Also, will python actually wait for `session.run_with_iobinding(io_binding)` to get a result from device before moving to the next line?
Im concerned that it might be executed async and python will only wait on the line that actually uses this output.

Reply:
I'd say just verify youre getting reasonable values for this (compared to driver perf) and feel free to resolve this

Reply:
I don't think its getting run async as I'm 

inception_v3, Rate = 107.53 QPS

vs our driver

Batch size: 1
Rate: 218.791 inferences/sec
Total time: 4.57057ms

Oddly we're slower. Maybe I am grabbing the time to transfer here as well

Reply:
ok its unlikely that you'll see it drop to half just because of transfers. Look at my other comment about averaging. And also this is probably unreliable because you're measuring latency on a "cold start run". Even when you average, make sure to run it once (or maybe even n times, Im not sure about the inner workings of ort) before recording the measurements

Reply:
Yeah I think its cold start.

inception_v3, Rate = 206.74 QPS

After doing an additional run before taking the measurement. Maybe its worth tracking setup time between runs.?

Reply:
I added a loop to run this 100 times like how we do a migraphx-driver perf. Looks like it gets a better measurement slightly slower (due to the overhead) than our migraphx-driver perf runs on the model.